shared/middleware/helpers/table.rb

```diff
@@ -140,12 +232,47 @@ def initialize(channel_id, storage_id, table_name)
     @table_name = table_name
 
     @hash = "#{@channel_id}:#{@table_name}:#{@storage_id}"
+    @metadata_hash = "#{@channel_id}:#{@table_name}:#{@storage_id}:metadata"
   end
```

Comment:
The previous comment may be irrelevant in light of my other suggestion to remove storage_id from the metadata_hash